lib/mixlib/install/options.rb

```diff
@@ -77,6 +75,15 @@ def initialize(options)
         # Store original options in cases where we must remap
         @original_platform_version = options[:platform_version]
 
+        # Eval extra products definition
+        extra_products = ENV.fetch("EXTRA_PRODUCTS_FILE", nil)
+        unless extra_products.nil?
+          PRODUCT_MATRIX.instance_eval(::File.read(extra_products), extra_products)
+        end
```

Comment:
I'm not sure how we would use this in practice on the Cinc side. As it stands now, it would require that environment variable to be set so that it would work. I wonder if there is a better way to handle this so that it works seamlessly in practice.

However, with the inclusion of the new `api_url` parameter, we could adjust our version of this to just add ours using our URLs I think.

Reply:
On cinc client we could just add the file for cinc products and define the env variable to this file path in the dist.rb file.

Using an environment variable allow to use omnitruck API with extra products by setting the var too.

With this we may update the mixlib-install in any chef-client with recent enough ruby (2.5 because of chef-utils ?) and add the environment variable before calling the omnitruck-updater cookbook to migrate from chef to cinc or cinc to chef (as actual cinc clients have no idea of chef products).

Reply:
I'd like to see how this works on our end before we commit to it. Can you do that please?

Reply:
So to summarize our talk on slack: That's not really easy to do from this PR as using it means cutting off on our overriding of mixlib-install and thus removing the tweaks we make on Gemfile to use cinc's version of mixlib install.

This change would allow using chef's mixlib-install gem and provide an extra file along with an environment variable in workstation commands to extend the amount of products handled.

My main interest in that is stopping to monkey patch either chef or cinc mixlib-install gem to address both products in the evolution I'm doing on omnitruck software for internal use.
It's a fork as Chef as been super clear about not handling alternative distributions on their endpoints, not even with a redirect, so no PR to come to upstream omnitruck to handle that, but as most things are handled by mixlib-install, that's be cool to avoid ugly patching and have a native way to do it.

Reply:
OK @Tensibai provided [this example](https://gitlab.com/cinc-project/upstream/chef/-/merge_requests/17/diffs) as a use case and it makes sense to me. So this has my approval now.

Reply:
Exemple implementation on cinc side: https://gitlab.com/cinc-project/upstream/chef/-/merge_requests/17